### PR TITLE
Port the `ImageClassifier` guide to `keras_core`

### PR DESCRIPTION
**Pull request by tirthasheshpatel**
Continuing #273.

> This PR ports the [ImageClassifier guide on keras.io](https://keras.io/guides/keras_cv/classification_with_keras_cv/) to keras_core. It uses the keras_cv.backend module to import Keras which provides support for both tf.keras and keras_core in a backend-agnostic manner.
> 
> To run the guide on Colab, use the [core branch from Ian's fork](https://github.com/ianstenbit/keras-cv/tree/core) of Keras-CV where we are porting backbones to keras_core.
> 
> cc @ianstenbit

### CODE REVIEW

File: examples/keras_io/vision/image_classifier.py [hunk 1]
```diff
@@ -26,14 +22,14 @@
...
 
-We use Professor Keras, the official Keras mascot, as a
-visual reference for the complexity of the material:
-
-![](https://storage.googleapis.com/keras-nlp/getting_started_guide/prof_keras_evolution.png)
-
 ## Multi-Backend Support
 
-Keras-CV's `ImageClassifier` model supports several backends like Jax, PyTorch, and TensorFlow with the help of `keras_core`. To enable multi-backend support in Keras-CV, set the `KERAS_CV_MULTI_BACKEND` environment variable. We can then switch between different backends by setting the `KERAS_BACKEND` environment variable. Currently, `tensorflow`, `jax`, and `torch` are supported.
+KerasCV's `ImageClassifier` model supports several backends like Jax, PyTorch,
```
Comment (fchollet) [inline examples/keras_io/vision/image_classifier.py:27]: JAX

File: examples/keras_io/vision/image_classifier.py [hunk 2]
```diff
@@ -26,14 +22,14 @@
...
+and TensorFlow with the help of `keras_core`. To enable multi-backend support
+in KerasCV, set the `KERAS_CV_MULTI_BACKEND` environment variable. We can
+then switch between different backends by setting the `KERAS_BACKEND`
+environment variable. Currently, `tensorflow`, `jax`, and `torch` are
```
Comment (fchollet) [inline examples/keras_io/vision/image_classifier.py:31]: Use backticks + double quotes for strings

File: examples/keras_io/vision/image_classifier.py [hunk 3]
```diff
@@ -0,0 +1,692 @@
...
+"""
+
+cut_mix_or_mix_up = keras_cv.layers.RandomChoice(
+    [cut_mix, mix_up], batchwise=True
+)
+augmenters += [cut_mix_or_mix_up]
+
+"""Now let's apply our final augmenter to the training data:"""
+
+augmenter = tf.keras.Sequential(augmenters)
```
Comment (fchollet) [inline examples/keras_io/vision/image_classifier.py:480]: There should be no tf.keras at all

File: examples/keras_io/vision/image_classifier.py [hunk 4]
```diff
@@ -0,0 +1,692 @@
...
+
+num_classes = dataset_info.features["label"].num_classes
+
+resizing = keras_cv.layers.Resizing(
+    IMAGE_SIZE[0], IMAGE_SIZE[1], crop_to_aspect_ratio=True
+)
+
+
+def preprocess_inputs(image, label):
+    image = tf.cast(image, tf.float32)
```
Comment (fchollet) [inline examples/keras_io/vision/image_classifier.py:167]: Since the example has quite a bit of TF-only APIs...

1. We could reduce or remove the TF dependency and keep it in the backend-agnostic folder.
2. We could move it to the TF-only folder.

File: examples/keras_io/vision/image_classifier.py (hunk 3 above)
Comment (ianstenbit) [reply examples/keras_io/vision/image_classifier.py:480]: Let's use the `Augmenter` which we're adding back to KerasCV in https://github.com/keras-team/keras-cv/pull/1978

File: examples/keras_io/vision/image_classifier.py (hunk 4 above)
Comment (ianstenbit) [reply examples/keras_io/vision/image_classifier.py:167]: The TF components are just for preprocessing which for now has to live in tf.data. I'm fine with moving it to TF-only, but I would expect this to work fine with all backends since this is constrained to the tf.data parts of the example

Comment (tirthasheshpatel) [reply examples/keras_io/vision/image_classifier.py:167]: I was trying your suggestions out @fchollet and seems like we would still need the `tf` import for one hot encoding since `CategoryEncoding` doesn't accept `float32` labels (so, even if we use it, we'd still need to cast labels to integers).

Other than that, everything else works. Let me know if this is OK from your side!

**Review by fchollet — APPROVED** [review]
LGTM, thanks!

**Comment by fchollet** [comment]
Can you also open a PR for this on keras-team/keras-io?

**Comment by tirthasheshpatel** [comment]
Sure, I still had one change in flight but I'll submit a follow-up! I will also submit a PR on keras.io.

Thanks for the reviews @fchollet @ianstenbit!
